core/src/main/java/com/linecorp/armeria/internal/server/annotation/DefaultNamedTypeInfoProvider.java

```diff
@@ -193,19 +196,22 @@ private FieldInfo fieldInfos(JavaType javaType, String name, String internalName
                                  Function<JavaType, List<FieldInfo>> childFieldsResolver) {
         TypeSignature typeSignature = toTypeSignature(fieldType);
         final FieldRequirement fieldRequirement;
-        if (typeSignature.isOptional()) {
-            typeSignature = typeSignature.typeParameters().get(0);
-            if (typeSignature.namedTypeDescriptor() instanceof Class) {
-                //noinspection OverlyStrongTypeCast
-                fieldType = mapper.constructType((Class<?>) typeSignature.namedTypeDescriptor());
+        if (typeSignature.type() == TypeSignatureType.OPTIONAL) {
+            typeSignature = ((ContainerTypeSignature) typeSignature).typeParameters().get(0);
```

Comment:
These APIs may be used only internally or by a few advanced users. For those uses, it looks good. However, there is no relationship between `TypeSignatureType.OPTIONAL` and `ContainerTypeSignature`, and users should check the implementation detail to know what `TypeSignature` is returned. Would it be worth making `TypeSignature.ofOptional()` return `ContainerTypeSignature` for clarification? For example:
```java
ContainerTypeSignature optional = TypeSignature.ofOptional(...);
```